### PR TITLE
Add information about library version into User-Agent header

### PR DESCRIPTION
**Pull request by dvacca**
Changing User-Agent header provided at each request into `onfido-java/{version}` to make it similar to format used by other client libraries. It includes a change to CI to run checks also on branches with name starting with  `release/`.

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -5,9 +5,13 @@ name: Java CI with Maven
 
 on:
   push:
-    branches: [ master ]
+    branches:
+      - master
+      - 'release/**'
```
Comment (DavidMealha) [inline .github/workflows/maven.yml:10]: I don't recall ever using release branches in client libraries, why is this change necessary?

Comment (dvacca) [reply .github/workflows/maven.yml:10]: It's just to avoid changing this file again as done [here](https://github.com/onfido/onfido-java/pull/71/files) to retrofit a change in an older release. Naming of release branch can be changed, anyway it's just to have the possibility to test and deliver changes when they're not on the master branch (i.e. deliver a patch for an older release): we can still use master branch for the _normal_ changes on last release.

Comment (DavidMealha) [reply .github/workflows/maven.yml:10]: Ah ok, makes sense 👍

**Review by DavidMealha — APPROVED** [review]
